File: ProyectoV2022/Probando/Mapper/PacienteMapper.cs
using ServicioMedico.DTO;
using ServicioMedico.Entities;

namespace ServicioMedico.Mapper
{
    public class PacienteMapper : Profile
    {
        public PacienteMapper()
        {
            CreateMap<PacienteDTO, Paciente>();
            CreateMap<Paciente, PacienteDTO>();
            CreateMap<PacienteSearchFilter, Paciente>();
            CreateMap<Paciente, PacienteSearchFilter>();
        }
    }
}

File: ProyectoSpring/src/main/java/net/acc/customproyect/mapper/PacienteMapper.java
package net.acc.customproyect.mapper;

import java.util.List;

import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

import net.acc.customproyect.entities.Paciente;
import net.acc.customproyect.entitiesDTO.PacienteSearchFilter;
import net.acc.customproyect.entitiesDTO.PacienteDTO;

@Mapper(componentModel="spring")
public interface PacienteMapper {
	
	PacienteMapper INSTANCE = Mappers.getMapper(PacienteMapper.class);
	
	 PacienteDTO PacienteToPacienteDto(Paciente paciente);
	 Paciente PacienteDTOToPaciente(PacienteDTO pacienteDTO);
	 PacienteSearchFilter PacienteToPacienteSearchFilterDTO(Paciente paciente);
	 Paciente PacienteSearchFilterDTOToPaciente(PacienteSearchFilter pacienteDTO);
	 List<PacienteDTO> pacienteListToPacienteDTOList(List<Paciente> pacientes);
	 List<PacienteSearchFilter> pacienteListToPacienteSearchFilterDTOList(List<Paciente> pacientes);
}
